Assets/Scripts/Character/CharacterInput.cs

```diff
@@ -0,0 +1,63 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+public class CharacterInput : MonoBehaviour
+{
+    public Collider2D Area;
+    public CharacterMovement Character;
+    public Collider2D[] areas;
+    public Transform[] transPoints;
+    List<Vector3> targets;
+
+    void Start()
+    {
+        targets = new List<Vector3>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        var mousePos3D = Camera.main.ScreenToWorldPoint(Input.mousePosition);
+        Vector2 mousePos2D = new Vector2(mousePos3D.x, mousePos3D.y); // Cast the mouse position to 2D
+        
+        if (Input.GetMouseButtonUp(0) && !(GameController.controller.isUI) && GameController.controller.lastUITime != Time.time)
```

Comment:
Path calculation should be done in separate method, so it could be called from other components, not just by clicking mouse button.